src/raw-to-zarr-to-bq.py

```diff
@@ -0,0 +1,206 @@
+
+
+def perform_data_operations(z_file: str, table: str, region: str, start_date: str,
+                            end_date: str, init_date: str):
+    # Function to process a single pair of z_file and table
+    try:
+        logger.info(f"Resizing zarr file: {z_file} started.")
+        resize_zarr_target(z_file, end_date, init_date)
+        logger.info(f"Resizing zarr file: {z_file} completed.")
+        logger.info(f"Data ingesting for {z_file} is started.")
+        ingest_data_in_zarr_dataflow_job(z_file, region, start_date, end_date, init_date,
+                                         project=PROJECT, bucket=BUCKET, python_path=PYTHON_PATH)
+        logger.info(f"Data ingesting for {z_file} is completed.")
+        start = f' "start_date": "{start_date}" '
+        end = f'"end_date": "{end_date}" '
+        zarr_kwargs = "'{" + f'{start},{end}' + "}'"
+        # TODO(dabhis): Find a way to ingest AR data into BQ.
+        # logger.info(f"Data ingesting into BQ table: {table} started.")
+        # ingest_data_in_bigquery_dataflow_job(z_file, table, region, zarr_kwargs)
+        # logger.info(f"Data ingesting into BQ table: {table} completed.")
+    except Exception as e:
+        logger.error(
+            f"An error occurred in process_zarr_and_table for {z_file}: {str(e)}")
+
+
+if __name__ == "__main__":
+    try:
+        parsed_args, unknown_args = parse_arguments_raw_to_zarr_to_bq("Parse arguments.")
+
+        logger.info("Program is started.")
+        data_date_range = date_range(
+            dates_data["first_day_third_prev"], dates_data["last_day_third_prev"]
+        )
+
+        for env_var in os.environ:
+            if API_KEY_PATTERN.match(env_var):
+                api_key_value = os.environ.get(env_var)
+                API_KEY_LIST.append(api_key_value)
+
+        additional_content = ""
+        for count, secret_key in enumerate(API_KEY_LIST):
+            secret_key_value = get_secret(secret_key)
+            additional_content += f'parameters.api{count}\n\
+                api_url={secret_key_value["api_url"]}\napi_key={secret_key_value["api_key"]}\n\n'
+
+        update_config_file(DIRECTORY, FIELD_NAME, additional_content)
+        logger.info("Raw data downloading start.")
+        raw_data_download_dataflow_job()
+        logger.info("Raw data downloaded successfully.")
+
+        logger.info("Raw data Splitting start.")
+        data_splitting_dataflow_job(
+            dates_data['first_day_third_prev'].strftime("%Y/%m"))
+        logger.info("Raw data Splitting successfully.")
+
+        logger.info("Data availability check started.")
+        data_is_missing = True  # Initialize with a non-zero value
+        while data_is_missing:
+            data_is_missing = check_data_availability(data_date_range)
+            if data_is_missing:
+                logger.warning("Data is missing.")
+                raw_data_download_dataflow_job()
+        logger.info("Data availability check completed.")
+
```

Comment:
Let's make logging consistent. Have a log before & after of each step.
And also at the end have a log something like -- "Automatic update for ARCO-ERA5 completed for {month}."